### PR TITLE
Update naming conventions, porter workflow in vanilla workspace

### PR DESCRIPTION
**Pull request by marrobi**
# PR for issue #132 

## What is being addressed

As part of the InnerEye deployment vanilla workspace resources need to follow the agreed naming convention.

Switched TF to use environment variables and Azure CLI to simplify what's being passed to each bundle.

To test this the bundle needed to be installed using porter, hence the iteration on porter workflow and dev container update

## How is this addressed

-  Update resource naming, and add missing parameters in terraform
-  Update porter bundles
-  Update install workflow to improve dev experience, add porter to dev container

### CODE REVIEW

**Comment by marrobi** [comment]
Will look at #172 first.

File: workspaces/vanilla/.env.sample [hunk 1]
```diff
@@ -1,8 +1,8 @@
...
+workspace_id="006"
+address_space="10.2.6.0/24"
 location=westeurope
 AZURE_TENANT_ID=_CHANGEME_
 AZURE_SUBSCRIPTION_ID=_CHANGEME_
-AZURE_CLIENT_ID=_CHANGEME_
-AZURE_CLIENT_SECRET=_CHANGEME_
\ No newline at end of file
+AZURE_SERVICE_PRINCIPAL_CLIENT_ID=_CHANGEME_
+AZURE_SERVICE_PRINCIPAL_CLIENT_PASSWORD=_CHANGEME_
```
Comment (tompaana) [inline workspaces/vanilla/.env.sample:8]:
```suggestion
AZURE_CLIENT_ID=_CHANGEME_
AZURE_CLIENT_SECRET=_CHANGEME_
```

Comment (tompaana) [reply workspaces/vanilla/.env.sample:8]: As much as like the long version (self-documenting!), the official convention seem to have become this. See e.g., https://docs.microsoft.com/en-us/dotnet/api/azure.identity.environmentcredential

File: workspaces/vanilla/azure.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{
+  "schemaVersion": "1.0.0-DRAFT+b6c701f",
+  "name": "azure",
+  "created": "2021-06-03T11:31:05.7314113Z",
+  "modified": "2021-06-03T11:31:05.7314113Z",
+  "credentials": [
+    {
+      "name": "azure_service_principal_client_id",
+      "source": {
+        "env": "AZURE_SERVICE_PRINCIPAL_CLIENT_ID"
```
Comment (tompaana) [inline workspaces/vanilla/azure.json:10]:
```suggestion
        "env": "AZURE_CLIENT_ID"
```

File: workspaces/vanilla/azure.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+      }
+    },
+    {
+      "name": "azure_service_principal_password",
+      "source": {
+        "env": "AZURE_SERVICE_PRINCIPAL_CLIENT_PASSWORD"
```
Comment (tompaana) [inline workspaces/vanilla/azure.json:16]:
```suggestion
        "env": "AZURE_CLIENT_SECRET"
```

**Review by tompaana — APPROVED** [review]

File: workspaces/vanilla/.env.sample (hunk 1 above)
Comment (marrobi) [reply workspaces/vanilla/.env.sample:8]: :) I changed that, but reverted back as is what you had used. Will change back. :)

**Pull request merged**